openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -205,15 +208,21 @@ def award_program_certificates(self, username):
                 CredentialsApiConfig.current(),
                 User.objects.get(username=settings.CREDENTIALS_SERVICE_USERNAME)  # pylint: disable=no-member
             )
-        except Exception, exc:  # pylint: disable=broad-except
+        except Exception as exc:  # pylint: disable=broad-except
             LOGGER.exception('Failed to create a credentials API client to award program certificates')
             # Retry because a misconfiguration could be fixed
-            raise self.retry(exc=exc)
+            raise self.retry(exc=exc, countdown=countdown, max_retries=config.max_retries)
 
+        retry = False
         for program_id in new_program_ids:
             try:
                 award_program_certificate(credentials_client, username, program_id)
                 LOGGER.info('Awarded certificate for program %s to user %s', program_id, username)
             except Exception:  # pylint: disable=broad-except
-                # keep trying to award other certs.
+                # keep trying to award other certs, but retry the whole task to fix any missing entries
                 LOGGER.exception('Failed to award certificate for program %s to user %s', program_id, username)
+                retry = True
+
+        if retry:
+            LOGGER.info('Retrying task to award failed certificats to user', username)
```

Comment:
`'certificats'` -> `'certificates'` (or [feel free to leave it](http://www.language-stickers.co.uk/Images/Certificate%20-%20French,%20cat.JPG))